v0/destinations/freshsales/utils.js

```diff
@@ -39,44 +81,284 @@ const getUserAccountDetails = async (userEmail, Config) => {
+    sales_activity_type_id: salesActivityTypeId
+  };
+  return responseBody;
+};
+
+/*
+ * This function is used for creating sales Activity of Contact.
+ * @param {*} payload - created after transformation
+ * @param {*} message - input message
+ * @param {*} Config - headers, apiKey...
+ * ref: https://developers.freshworks.com/crm/api/#list_all_sales_activities
+ */
+const UpdateContactWithSalesActivity = async (payload, message, Config) => {
+  const requestOptions = {
+    headers: {
+      Authorization: `Token token=${Config.apiKey}`,
+      "Content-Type": "application/json"
+    }
+  };
+  if (!payload.sales_activity_name && !payload.sales_activity_type_id) {
+    throw new CustomError(
+      `Either of sales activity name or sales activity type id is required. Aborting!`,
+      400
+    );
+  }
+
+  const email = get(message, "email");
+
+  if (!payload.targetable_id && !email) {
+    throw new CustomError(
+      `Either of email or targetable_id is required for creating sales activity. Aborting!`,
+      400
+    );
+  }
+
+  if (payload.sales_activity_type_id) {
+    let responseBody;
+    if (payload.targetable_id) {
+      responseBody = {
+        ...payload,
+        sales_activity_type_id: payload.sales_activity_type_id
+      };
+    } else {
+      responseBody = responseBuilderWithContactDetails(
+        email,
+        Config,
+        payload,
+        payload.sales_activity_type_id
+      );
+    }
+    return responseBody;
+  }
+  // with sales activity name
+  const endPoint = `https://${Config.domain}${CONFIG_CATEGORIES.SALES_ACTIVITY.baseUrlListAll}`;
+  let salesActivityResponse = await httpGET(endPoint, requestOptions);
+  salesActivityResponse = processAxiosResponse(salesActivityResponse);
+  if (salesActivityResponse.status !== 200) {
+    const errMessage = salesActivityResponse.response.errors?.message || "";
+    const errorStatus = salesActivityResponse.response.errors?.code || "500";
+    throw new CustomError(
+      `failed to fetch sales activities details ${errMessage}`,
+      errorStatus
+    );
+  }
+
+  const salesActivityList = salesActivityResponse.response.sales_activity_types;
+  if (salesActivityList && Array.isArray(salesActivityList)) {
+    const listDetails = salesActivityList.find(
+      list => list.name === payload.sales_activity_name
+    );
+    if (!listDetails) {
+      throw new CustomError(
+        `sales Activity ${payload.sales_activity_name} doesn't exists. Aborting!`,
+        400
+      );
+    }
+    let responseBody;
+    if (payload.targetable_id) {
+      responseBody = {
+        ...payload,
+        targetable_id: payload.targetable_id,
+        sales_activity_type_id: listDetails.id
+      };
+    } else {
+      responseBody = await responseBuilderWithContactDetails(
+        email,
+        Config,
+        payload,
+        listDetails.id
+      );
+    }
+
+    return responseBody;
+  }
+  throw new CustomError(`sales activity types must be an array.`, 400);
```

Comment:
Unable to fetch correct list of sales activity

Reply:
updated